src/js/view/floating/floating-controller.ts

```diff
@@ -0,0 +1,303 @@
+import type Model from 'controller/model';
+import type Preview from 'view/preview';
+import type View from 'view/view';
+import type { FloatConfig } from 'controller/model';
+import FloatingDragUI from 'view/floating/floating-drag-ui';
+import { OS } from 'environment/environment';
+import { deviceIsLandscape } from 'utils/dom';
+import { isIframe } from 'utils/browser';
+import {
+    addClass,
+    removeClass
+} from 'utils/dom';
+import { STATE_IDLE, STATE_COMPLETE, STATE_ERROR } from 'events/events';
+import { isNumber } from 'utils/underscore';
+import {
+    style,
+} from 'utils/css';
+import viewsManager from 'view/utils/views-manager';
+
+const FLOATING_TOP_OFFSET = 62;
+let _floatingPlayer: HTMLElement | null = null;
+
+export default class FloatingController {
+    _playerEl: HTMLElement;
+    _parent: View;
```

Comment:
Could we avoid using a cyclical reference like this?
- pass in `playerBounds` when they change instead of depending on `parent.getPlayerBounds`
- dispatch events that the view listens to to call `responsiveListener` and `onAspectRatioChange` 
- remove the need to call `parent.getPlayerSizeStyles` by leaving the `updateFloatingSize` in view or handling with events (in `responsiveListener` , replace `floatingController.resize`)

Reply:
That way you don't have to worry about forgetting to set `_parent` to `null` in `destroy` and `floatingController` to null in the view after destroying it.